### PR TITLE
[PW-5712] - Check address issue on plugin v6

### PR DESCRIPTION
**Pull request by Morerice**
**Description**
This PR updates the regex used to identify the house number when it is entered in the same line as the street. Currently the houseNumber will be correctly sent to Adyen only if it is entered after the street. This PR will cater for the case where the number is also entered before the street.

Also when finalizing an invoice, check if the pay() method was called.

**Tested scenarios**
* House number entered before street
* House number entered after street

### CODE REVIEW

File: Helper/Invoice.php [hunk 1]
```diff
@@ -118,7 +118,7 @@ public function finalizeInvoices(Order $order, Notification $captureNotification
         foreach ($invoiceCollection as $invoice) {
             $parsedTransId = $this->adyenDataHelper->parseTransactionId($invoice->getTransactionId());
             // Loose comparison based on how Magento does the comparison in entity
-            if ($invoice->getState() == InvoiceModel::STATE_OPEN) {
+            if ($invoice->getState() == InvoiceModel::STATE_OPEN || !$invoice->wasPayCalled()) {
```
Comment (acampos1916) [inline Helper/Invoice.php:121]: This probably is unrelated to the PR, should we mention it in the PR description?

**Review by acampos1916 — APPROVED** [review]

File: Helper/Invoice.php (hunk 1 above)
Comment (Morerice) [reply Helper/Invoice.php:121]: Updated description

**Review by peterojo — APPROVED** [review]
